### PR TITLE
Improve shutdown logic of language server

### PR DESCRIPTION
**Pull request by hubertp**
### Pull Request Description

This PR addresses problems mentioned in #7470 and #7729:
- shutting a language server explicitly will not lead to a soft shutdown
- `project/status` endpoint returns the state of the language server

`LanguageServerController` now also signed up for `ClientConnect` messages. For it to be unambiguous, we need to carry around the port number of the language server as a way of identifying the right one.

One can now use `project/status` to additionally determine the state of the language server.

Also relies on a proper fix for #7765.

### Important Notes

Relies on a proper fix for #7765.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: CHANGELOG.md
Comment (JaroslavTulach) [inline CHANGELOG.md:960]: I don't see change in the language server protocol specification. When at it, can you also start versioning the protocol? At least include semantic version to the header of the document and mention that `project/status` message is available _since_ that version.

Here is an [apichanges inspiration](https://bits.netbeans.org/16/javadoc/org-openide-util-lookup/apichanges.html).

**Review by JaroslavTulach — DISMISSED** [review]
I cannot comment on the Akka implementation, but I've noticed missing documentation and versioning.

File: lib/scala/json-rpc-server/src/main/scala/org/enso/jsonrpc/JsonRpcServer.scala
Comment (Akirathan) [inline lib/scala/json-rpc-server/src/main/scala/org/enso/jsonrpc/JsonRpcServer.scala]: Uncomment this line and provide a proper filter in test logging config? Or delete this line?

File: lib/scala/project-manager/src/main/resources/application.conf
Comment (Akirathan) [inline lib/scala/project-manager/src/main/resources/application.conf:120]: Are we certain that 3 seconds is sufficient? What is the reason for this particular constant?

File: lib/scala/project-manager/src/main/scala/org/enso/projectmanager/infrastructure/languageserver/LanguageServerController.scala
Comment (Akirathan) [inline lib/scala/project-manager/src/main/scala/org/enso/projectmanager/infrastructure/languageserver/LanguageServerController.scala:180]: Maybe add a ScalaDoc explaining that the `Int` here is port? It is clear in this PR, but might not be clear in 2 months.

File: lib/scala/project-manager/src/main/scala/org/enso/projectmanager/infrastructure/languageserver/LanguageServerGateway.scala
Comment (Akirathan) [inline lib/scala/project-manager/src/main/scala/org/enso/projectmanager/infrastructure/languageserver/LanguageServerGateway.scala]: Tuple of `(Boolean, Boolean)`? What is the other `Boolean`? pls update the docs

File: lib/scala/project-manager/src/main/scala/org/enso/projectmanager/requesthandler/ProjectStatusHandler.scala
Comment (Akirathan) [inline lib/scala/project-manager/src/main/scala/org/enso/projectmanager/requesthandler/ProjectStatusHandler.scala]: Whose initials are `RW`?

Comment (hubertp) [reply lib/scala/project-manager/src/main/scala/org/enso/projectmanager/requesthandler/ProjectStatusHandler.scala]: Radek, looks like a c&p typo, sorry

File: lib/scala/project-manager/src/main/resources/application.conf
Comment (hubertp) [reply lib/scala/project-manager/src/main/resources/application.conf:120]: Yes, it's typically a subsecond thing. I think I was initially paranoid and gave it a bit too much.

File: CHANGELOG.md
Comment (hubertp) [reply CHANGELOG.md:960]: I added documentation for the protocol but not the versioning. I don't think the latter has been fully agreed on and feels wrong to suddenly plug it into this PR without adding logic that would negotiate protocol between LS and IDE. I think it's a good idea in general just not in this PR which is already not trivial to review.

Besides, the current approach seems to be to "always use latest" and its not breaking the protocol, just adding a route.

Comment (4e6) [reply CHANGELOG.md:960]: We had a discussion about versioning #7639, and I think, instead of versioning the protocol, it would be reasonable to make IDE use the latest language server API (because they are supposed to work together), and just track the oldest engine version that the current language server supports.

**Review by 4e6 — APPROVED** [review]

File: CHANGELOG.md
Comment (JaroslavTulach) [reply CHANGELOG.md:960]: Compatibility is about **attitude**. Once we start our effort to be compatible we'll be (mostly) compatible. Until we start any effort (e.g. at least versioning of the protocol documentation!), we'll be **incompatible**.

> I don't think the latter has been fully agreed on

Let me know what kind of approval you need for starting the semantic versioning of the language server specification.

Comment (hubertp) [reply CHANGELOG.md:960]: > Let me know what kind of approval you need for starting the semantic versioning of the language server specification.

I don't want to add versioning in this PR as its not directly related to what this is trying to fix. The whole versioning story is part of the #7639 discussion and #7717 ticket. 

But I do need your approval, @JaroslavTulach, since I can't merge this fix otherwise.

**Pull request merged**
